### ISSUE TITLE
[SF][RadioButton]: [A11y]Required property is not read by Screen Reader JAWS

### ISSUE THREAD

**Issue by VidhiNPatel** [SAP/ui5-webcomponents#11786]
### Describe the bug

Setting the "required=true" attribute on the RadioButton component does not convey the required status to screen reader users.

### Reproduction steps

1. Go to https://sap.github.io/ui5-webcomponents-react/v2/?path=/docs/inputs-radiobutton--docs 
2. Run the JAWS
3. Check that the RadioButton component does not convey the required status to screen reader users 

### Expected Behaviour

All "Required" fields must be announced as "required" by screen readers

### UI5 Web Components for React Version

2.11.1

### UI5 Web Components Version

2.11.0

### Browser

Chrome

### Relevant log output

```shell

```

### Declaration

- [x] I’m not disclosing any internal or sensitive information.

**Comment by Lukas742** [comment]
Thanks for reporting! I'll forward this issue to our UI5 Web Components Colleagues as the affected component is developed in their repository.

**Comment by didip1000** [comment]
Hey @SAP/ui5-webcomponents-topic-rd,

Can you please take a look at this issue?

Best,
Diana

**Comment by GerganaKremenska** [comment]
@SAP/ui5-webcomponents-topic-rd Did some investigation and shared my findings with the team.

**Comment by TeodorTaushanov** [comment]
Hello,

To ensure proper screen reader accessibility, `ui5-radio-button` elements should be enclosed within a <div> that has the attribute role="radiogroup".

More info:
https://www.w3.org/WAI/ARIA/apg/patterns/radio/examples/radio/

Additionally, you can set aria-required="true" on this parent element. Here’s how your code should look:

```
<div role="radiogroup" aria-labelledby="radioGroupTitle1" aria-required="true">
	<ui5-title id="radioGroupTitle1">Select:</ui5-title>
	<ui5-radio-button value="optionA" name="myGroup" text="Option A"></ui5-radio-button>
	<ui5-radio-button value="optionB" name="myGroup" text="Option B"></ui5-radio-button>
	<ui5-radio-button value="optionC" name="myGroup" text="Option C"></ui5-radio-button>
</div>
```

If the radio buttons are part of a form, you can add a required attribute to any element in the radio group to ensure the form is validated before it is submitted:

```
<form>
  <div role="radiogroup" aria-labelledby="radioGroupTitle1" aria-required="true">
  	<ui5-title id="radioGroupTitle1">Select:</ui5-title>
  	<ui5-radio-button value="optionA" name="myGroup" text="Option A" required></ui5-radio-button>
  	<ui5-radio-button value="optionB" name="myGroup" text="Option B"></ui5-radio-button>
  	<ui5-radio-button value="optionC" name="myGroup" text="Option C"></ui5-radio-button>
  </div>
 <input type="submit">
</form>
```

Although it might seem unusual to include this parent element, it is necessary because this is how screen readers have been designed to work, despite it being different from form validation behavior.


Best,
Teodor

**Comment by shubh1502** [comment]
Hi, 

I did try the mentioned fix, still JAW's SR doesn't read it as required.
Attaching the screenshot for the same:

<img width="1892" height="1090" alt="Image" src="https://github.com/user-attachments/assets/372431d7-2277-45fe-9110-f608e59592f9" />

Let me know if anything else is required from my end.

**Comment by TeodorTaushanov** [comment]
@shubh1502, is there any way for us to test this? Alternatively, could you share your code here so we can take a closer look?

**Comment by shubh1502** [comment]
Hi @TeodorTaushanov , sure please find the attached SS for reference on how do we use used the suggestion in our code.

<img width="946" height="336" alt="Image" src="https://github.com/user-attachments/assets/79b8b389-26d6-46ad-83d0-9e17fd59a60b" />

After this i see on Dev Tools this is how it shows up.

<img width="1910" height="738" alt="Image" src="https://github.com/user-attachments/assets/1e16f674-3edb-48cf-8c19-4b0c6b3badf5" />

Please let me know anything else is required from my end.
Thanks

**Comment by TeodorTaushanov** [comment]
Hello @shubh1502,

I'm checking this code with JAWS 2026:
https://stackblitz.com/edit/js-yudpcfqz?file=index.js,index.html,package.json

```
<div
    aria-labelledby="radioGroupTitle1"
    role="radiogroup"
    aria-required="true"
  >
    <ui5-title id="radioGroupTitle1">Select:</ui5-title>
    <ui5-radio-button
      value="optionA"
      name="myGroup"
      text="Option A"
    ></ui5-radio-button>
    <ui5-radio-button
      value="optionB"
      name="myGroup"
      text="Option B"
    ></ui5-radio-button>
    <ui5-radio-button
      value="optionC"
      name="myGroup"
      text="Option C"
    ></ui5-radio-button>
  </div>
</div>
```

and here is the speech history:

```
To change the selection press Up or Down Arrow.
Select: group Required
Option A radio button not checked
1 of 3
```

Best,
Teo

**Comment by shubh1502** [comment]
Hi @TeodorTaushanov , Thanks for your inputs it actually helped.

In our case aria-labelledby was not possible as we did not had a particular id to link the label element.

But this gave a idea of directly using aria-label and it does works perfectly 👍🏻 

SS:

<img width="1788" height="1022" alt="Image" src="https://github.com/user-attachments/assets/5e6671ba-4634-4dbe-ad9e-b82c87b2259e" />

**Thanks for your help, we can not close the ticket.**

**Comment by TeodorTaushanov** [comment]
Great, glad we were able to resolve the issue.
